core/PRP/Geometry/plDrawableSpans.cpp

```diff
@@ -661,13 +661,11 @@ void plDrawableSpans::clearSpans()
     fSpanSourceIndices.clear();
 }
 
-size_t plDrawableSpans::addIcicle(const plIcicle& span)
+size_t plDrawableSpans::addIcicle(plIcicle* span)
 {
-    plIcicle* iceCopy = new plIcicle(span);
-    fIcicles.push_back(iceCopy);
-    size_t iceId = fIcicles.size() - 1;
-    fSpans.push_back(fIcicles[iceId]);
-    fSpanSourceIndices.push_back(iceId | kSpanTypeIcicle);
+    fIcicles.push_back(span);
+    fSpans.push_back(span);
+    fSpanSourceIndices.push_back((fSpans.size() - 1) | kSpanTypeIcicle);
```

Comment:
I assume this change to use the span index instead of the icicle index is intentional?

Reply:
I don't remember having a specific intention there. I'll change it since in libHSPlasma fSpans should effectively be a copy of fIcicles.